sbncode/SinglePhotonAnalysis/HelperFunctions/helper_math.h

```diff
@@ -0,0 +1,295 @@
+
+  // right, but can be simplified
+        double d2 = pow(x1-x0,2)+pow(y1-y0,2)+pow(z1-z0,2)+2*t*((x2-x1)*(x1-x0)+(y2-y1)*(y1-y0)+(z2-z1)*(z1-z0))+t*t*( pow(x2-x1,2)+pow(y2-y1,2)+pow(z2-z1,2));
+
+        return sqrt(d2);
+    }
+
+    // minimum distance between point and dead wire
+    double distanceToNearestDeadWire(int plane, double Ypoint, double Zpoint,
+                const geo::GeometryCore * geom,
+                std::vector<std::pair<int,int>> & bad_channel_list_fixed_mcc9 ){
+
+             double min_dist = 999999;
+            
+             for(size_t i=0; i< bad_channel_list_fixed_mcc9.size(); i++){
+                   int channel = bad_channel_list_fixed_mcc9[i].first;                                       
+                   int is_ok = bad_channel_list_fixed_mcc9[i].second;       
+                        if(is_ok>1)continue;
+
+                   auto wireids = geom->ChannelToWire(channel); //type of wireids: IDs of all the connected wires to the channel
+                   auto result = geom->WireEndPoints(wireids[0]);
+                        
+                    //std::cout<<"KNK: "<<bc<<" "<<hs[0]<<" "<<result.start().X()<<" "<<result.start().Y()<<" "<<result.start().Z()<<" "<<result.end().X()<<" "<<result.end().Y()<<" "<<result.end().Z()<<std::endl; 
+//                    std::cout<<wireids[0].Plane<<" "<<result.start().X()<<std::endl;
+                    if(plane != (int)wireids[0].Plane) continue;
+
+        // for the dead wires on the same plane 
+                    std::vector<double> start = {0.0,result.start().Y(),result.start().Z()};
+                    std::vector<double> end = {0.0,result.end().Y(),result.end().Z()};
+                    std::vector<double> point = {0.0,Ypoint,Zpoint};
+                    double dist = dist_line_point(start,end,point);
+                    min_dist = std::min(dist,min_dist);
+             }
+                 
+    return min_dist;
+
+    }
+
+//--------------- end of copying from bad_channel_matching.h
+
+    double impact_paramater_shr(double x, double y, double z, art::Ptr<recob::Shower> & shr){
+
+        std::vector<double> vert = {x,y,z}; 
+        std::vector<double> start = {shr->ShowerStart().X(), shr->ShowerStart().Y(),shr->ShowerStart().Z()};
+        std::vector<double> abit = {shr->ShowerStart().X() + shr->Direction().X(),  shr->ShowerStart().Y()+shr->Direction().Y(),  shr->ShowerStart().Z()+shr->Direction().Z()};
+
+        return dist_line_point(start, abit, vert);
+
+    }
+
+    // invariant mass of a particle that decays to two showers
+    double  implied_invar_mass(double vx, double vy, double vz, art::Ptr<recob::Shower> & s1, double E1,  art::Ptr<recob::Shower> &s2, double E2){
+
+        double s1x = s1->ShowerStart().X()-vx;
+        double s1y = s1->ShowerStart().Y()-vy;
+        double s1z = s1->ShowerStart().Z()-vz;
+        double norm1  = sqrt(pow(s1x,2)+pow(s1y,2)+pow(s1z,2));
```

Comment:
Use `std:::pythag(s1x, s1y, s1z)`

Reply:
Apply the update with `std::hypot(s1x,s1y,s1z)`